### PR TITLE
docs(blog): improve strapi-custom-api to fix some best practices

### PR DESCRIPTION
**Pull request by derrickmehaffy**
## Description

Hello from the Strapi team o/

Read through your post and I noticed a few things that could be done a bit better so figured I'd do a PR. Note in my opinion I don't think it's really needed to create a new API but didn't want to completely rewrite the latter half of your article. Generally you could just place the controller + service in the generated factory files for the content-type and create a new route in that same content-type.

That being said you don't actually need to write any code to do what you've described here as you could just do a normal filter for this instead:

`GET /api/reading-list?filters[keyword][$eq]=yourKeyword` 

But I get wanting to get started with Strapi so what you are doing isn't wrong per-say but just wanted to let you know that what you want to do is doable without writing any code :wink:

### CODE REVIEW

**Comment by njfamirm** [comment]
Hi dear friend!
With thanks for your attention
Yes, I will definitely check

**Comment by njfamirm** [comment]
> `GET /api/reading-list?filters[keyword][$eq]=yourKeyword`

I saw this `filter` but I didn't know how to search in the list of keywords! maybe it's better to save the keywords as a list in db, but I didn't manage to do that.
If I can solve this case, I will switch to the filter

Of course, my goal in this tutorial was to talk about custom API. 
Maybe I need to add another idea to include this goal in the post 😅

**Comment by derrickmehaffy** [comment]
> > `GET /api/reading-list?filters[keyword][$eq]=yourKeyword`
> 
> I saw this `filter` but I didn't know how to search in the list of keywords! maybe it's better to save the keywords as a list in db, but I didn't manage to do that. If I can solve this case, I will switch to the filter
> 
> Of course, my goal in this tutorial was to talk about custom API. Maybe I need to add another idea to include this goal in the post 😅

Hahaha, indeed there are a lot of use-cases for it. I've done several best practice talks if you want to watch some of them here is our unlisted Youtube Playlist: https://www.youtube.com/playlist?list=PL7Q0DQYATmvg5Ip7hpXypLf_G3-zEWhR6

Typically when I see entirely custom routes + controllers + services being extremely useful is when you are talking to external services but not enough to justify a plugin. Alternatively when you need to condense a response for multiple content-types it can be very useful.

If you are ever on our Discord (https://discord.gg/strapi) we do daily office hours M-F 10:30am to 11:30am PST so you are always welcome to come hang out and chat with us. The Discord is also a really good source of questions and topics if you want to write more blog posts as we can never have enough community members helping share knowledge.

We love to see community members writing articles like this and we even have our own "write for the community" program too :wink:

**Comment by njfamirm** [comment]
yes, thank you!
Your importance to the community is really admirable! I have rarely seen such a case!

File: site/en/blog/strapi-custom-api.md [hunk 1]
```diff
@@ -197,7 +190,7 @@ export default {search};
 
 Requests sent to Strapi on any URL are handled by routes. <sup> <a href="https://docs.strapi.io/dev-docs/backend-customization/routes" target="_blank">ReadMore</a></sup>
 
-We must add this endpoint and sets the previous controller as the route handler, put this code to `api/reading-list-search/config/routes.json`:
```
Comment (njfamirm) [inline site/en/blog/strapi-custom-api.md:200]: Wowwwww, what a funny mistake 🤣🤣🤣

File: site/en/blog/strapi-custom-api.md [hunk 2]
```diff
@@ -151,21 +151,15 @@ In this step, we use the service to create a function to get the reading list an
...
-    let filteredEntries = entries;
-    if (keyword) {
-      filteredEntries = entries.filter((entry) => {
-        const keywords = entry.keyword.split('\n').map((keyword) => keyword.trim());
-        return keywords.includes(keyword);
-      });
-    }
-
-    return filteredEntries;
+    return await strapi.entityService.findMany('api::reading-list.reading-list', { filters: { keyword: { $eq: keyword }}});
```
Comment (njfamirm) [inline site/en/blog/strapi-custom-api.md:160]: All items are correct, thank you

Only this line can't search in a list, I divide the keywords based on the new line, can handle it in other way?

**Review by njfamirm — CHANGES_REQUESTED** [review]

**Comment by njfamirm** [comment]
If you don't have time to continue, tell me so that I can continue

**Review by njfamirm — APPROVED** [review]
I'm continue this

**Comment by derrickmehaffy** [comment]
Sorry about that! Didn't see your requested changes, there are ways to do multi-filter using the $or operator if you wanted to dive down that path but either way is fine honestly, generally we try to avoid doing filtering in JavaScript since it'll be slower than doing it at the database level.

**Comment by njfamirm** [comment]
> Sorry about that! Didn't see your requested changes, there are ways to do multi-filter using the $or operator if you wanted to dive down that path but either way is fine honestly, generally we try to avoid doing filtering in JavaScript since it'll be slower than doing it at the database level.

you're welcome
Thank you, I will check

**Pull request merged**